src/Symfony/Component/JsonEncoder/CacheWarmer/EncoderDecoderCacheWarmer.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonEncoder\CacheWarmer;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface;
+use Symfony\Component\JsonEncoder\DataModel\Decode\DataModelBuilder as DecodeDataModelBuilder;
+use Symfony\Component\JsonEncoder\DataModel\Encode\DataModelBuilder as EncodeDataModelBuilder;
+use Symfony\Component\JsonEncoder\Decode\DecodeFrom;
+use Symfony\Component\JsonEncoder\Decode\DecoderGenerator;
+use Symfony\Component\JsonEncoder\Encode\EncodeAs;
+use Symfony\Component\JsonEncoder\Encode\EncoderGenerator;
+use Symfony\Component\JsonEncoder\Exception\ExceptionInterface;
+use Symfony\Component\JsonEncoder\Mapping\PropertyMetadataLoaderInterface;
+use Symfony\Component\TypeInfo\Type;
+
+/**
+ * Generates encoders and decoders PHP files.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @internal
+ */
+final class EncoderDecoderCacheWarmer implements CacheWarmerInterface
+{
+    private EncoderGenerator $encoderGenerator;
+    private DecoderGenerator $decoderGenerator;
+
+    /**
+     * @param list<class-string>   $encodableClassNames
+     * @param array<string, mixed> $encoderConfig
+     * @param array<string, mixed> $decoderConfig
+     */
+    public function __construct(
+        private array $encodableClassNames,
+        PropertyMetadataLoaderInterface $encodePropertyMetadataLoader,
+        PropertyMetadataLoaderInterface $decodePropertyMetadataLoader,
+        string $encodersDir,
+        string $decodersDir,
+        private array $encoderConfig,
+        private array $decoderConfig,
+        private LoggerInterface $logger = new NullLogger(),
+    ) {
+        $this->encoderGenerator = new EncoderGenerator(new EncodeDataModelBuilder($encodePropertyMetadataLoader), $encodersDir);
+        $this->decoderGenerator = new DecoderGenerator(new DecodeDataModelBuilder($decodePropertyMetadataLoader), $decodersDir);
```

Comment:
Actually, `EncoderGenerator` and `DecoderGenerator` are internal classes for the moment, as they are internal helpers. Therefore it would be impossible to do a `new EncoderDecoderCacheWarmer(...)` (the same goes for `new JsonEncoder(...)` and `new JsonDecoder(...)`).

We can rethink about it whenever we want for some reason to expose the generators. WDYT?